src/mon/MgrMonitor.cc

```diff
@@ -1051,7 +1051,13 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
 
   if (prefix == "mgr fail") {
     string who;
-    cmd_getval(cmdmap, "who", who);
+    if (!cmd_getval(cmdmap, "who", who)) {
+      if (!map.active_gid) {
+	ss << "Currently no active mgr";
+	goto out;
```

Comment:
shall we set `r` before bailing out? like
```c++
r = -EINVAL;
```
or
```c++
r = -ENOENT;
```
?

Reply:
I think in order to be idempotent we should succeed if there is no active mgr...